# HG changeset patch
# User Jacob Dawid <jacob.dawid@googlemail.com>
# Date 1337701605 -7200
#      Tue May 22 17:46:45 2012 +0200
# Branch gui
# Node ID 7b607001d5714c6f4dc90b9c9b199fcbb9f37933
# Parent  f346343654a49605005745e72fb79a637425b8e5
Copy and paste into the terminal window works now (ctrl+c, ctrl+v).

* MainWindow (construct): Added connects and shortcuts.

diff --git a/gui/src/MainWindow.cpp b/gui/src/MainWindow.cpp
--- a/gui/src/MainWindow.cpp
+++ b/gui/src/MainWindow.cpp
@@ -245,24 +245,27 @@ MainWindow::construct ()
   QAction *settingsAction = fileMenu->addAction (tr ("Settings"));
   fileMenu->addSeparator ();
   QAction *exitAction = fileMenu->addAction (tr ("Exit"));
 
   QMenu *editMenu = menuBar ()->addMenu (tr ("&Edit"));
   QAction *cutAction
       = editMenu->addAction (QIcon::fromTheme ("edit-cut",
         style->standardIcon (QStyle::SP_FileIcon)), tr ("Cut"));
+  cutAction->setShortcut(QKeySequence(Qt::CTRL + Qt::Key_X));
 
   QAction *copyAction
       = editMenu->addAction (QIcon::fromTheme ("edit-copy",
         style->standardIcon (QStyle::SP_FileIcon)), tr ("Copy"));
+  copyAction->setShortcut(QKeySequence(Qt::CTRL + Qt::Key_C));
 
   QAction *pasteAction
       = editMenu->addAction (QIcon::fromTheme ("edit-paste",
         style->standardIcon (QStyle::SP_FileIcon)), tr ("Paste"));
+  pasteAction->setShortcut(QKeySequence(Qt::CTRL + Qt::Key_V));
 
   QAction *undoAction
       = editMenu->addAction (QIcon::fromTheme ("edit-undo",
         style->standardIcon (QStyle::SP_FileIcon)), tr ("Undo"));
 
   QAction *redoAction
       = editMenu->addAction (QIcon::fromTheme ("edit-redo",
         style->standardIcon (QStyle::SP_FileIcon)), tr ("Redo"));
@@ -330,16 +333,19 @@ MainWindow::construct ()
 
   connect (m_filesDockWidget, SIGNAL (openFile (QString)), this, SLOT (openExistingFile (QString)));
   connect (m_historyDockWidget, SIGNAL (information (QString)), this, SLOT (reportStatusMessage (QString)));
   connect (m_historyDockWidget, SIGNAL (commandDoubleClicked (QString)), this, SLOT (handleCommandDoubleClicked (QString)));
   connect (saveWorkspaceAction, SIGNAL (triggered ()), this, SLOT (handleSaveWorkspaceRequest ()));
   connect (loadWorkspaceAction, SIGNAL (triggered ()), this, SLOT (handleLoadWorkspaceRequest ()));
   connect (clearWorkspaceAction, SIGNAL (triggered ()), this, SLOT (handleClearWorkspaceRequest ()));
 
+  connect (copyAction, SIGNAL (triggered()), m_terminalView, SLOT(copyClipboard ()));
+  connect (pasteAction, SIGNAL (triggered()), m_terminalView, SLOT(pasteClipboard ()));
+
   setWindowTitle ("Octave");
 
   addDockWidget (Qt::LeftDockWidgetArea, m_workspaceView);
   addDockWidget (Qt::LeftDockWidgetArea, m_historyDockWidget);
   addDockWidget (Qt::RightDockWidgetArea, m_filesDockWidget);
   setStatusBar (m_statusBar);
 
   // this has to be done only once, not for each editor
